i2c_connector.py

```diff
@@ -0,0 +1,85 @@
+from smbus2 import SMBus
+
+
+class I2C:
+    """ Represents the I2C connection and all its
+    functions necessary to create the communication
+    between the Arduino and the Raspberry Pi. """
+
+    def __init__(self, address, number_of_ldrs=2):
+        self.connector = SMBus(1)
+        self.SLAVE_ADDRESS = address
+        self.number_of_ldrs = number_of_ldrs
+
+    def get_arduino_data(self):
+        """ Fetchs the Arduino data transfered through I2C. """
+        return self.connector.read_i2c_block_data(self.SLAVE_ADDRESS, 1,
+                                                  (self.number_of_ldrs * 2))
+
+    def get_ldr_values(self):
+        """ Gets the ldr values and validates it. """
+        brute_data = self.get_arduino_data()
+        ldr_values = convert_byte_to_integer(brute_data)
+        validates_ldr_data(ldr_values)
+        return ldr_values
+
+
+def convert_byte_to_integer(data):
+    """ Converts the byte data to an integer. """
+    ldr_data = []
+    for i in range(int(len(data)/2)):
+        # Creates a list with the ldrs value as integers
+        ldr_data.append(aggregate_bytes(data[i*2], data[i*2 + 1]))
+    return ldr_data
+
+
+def aggregate_bytes(most_representative, least_representative):
+    """ Aggregates n bytes into a single word. """
+    return (most_representative << 8 | least_representative) & 0xffffffff
+
+
+def check_ldr_list_length(ldr_list):
```

Comment:
Estranho ter uma função que só joga exceções, ainda mais se chamando de check_ldr.... Se for ver, ela não retorna nada? Tu poderia então ter só uma função "assert_ldr_data_is_valid(ldr_list)" e fazer todas verificações nessa função - ai sim faz certo sentido ela retornar void (nada) e só jogar exceções. Além disso seria interessante